### PR TITLE
Go version 1.14

### PR DESCRIPTION
**Pull request by aweris**
Go version bump.

Solves: https://github.com/prometheus-operator/kube-prometheus/issues/493

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,4 +50,4 @@ $(BIN_DIR):
 
 $(TOOLING): $(BIN_DIR)
 	@echo Installing tools from scripts/tools.go
-	@cat scripts/tools.go | grep _ | awk -F'"' '{print $$2}' | GOBIN=$(BIN_DIR) xargs -tI % go install %
+	@cd scripts && cat tools.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go build -modfile=go.mod -o $(BIN_DIR) %
```
Comment (kakkoyun) [inline Makefile:53]: 👍

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ on:
   - push
   - pull_request
 env:
-  golang-version: '1.13'
+  golang-version: '1.14'
```
Comment (kakkoyun) [inline .github/workflows/ci.yaml:6]: Why not migrate to go `1.15`?

Comment (aweris) [reply .github/workflows/ci.yaml:6]: it's possible to bump `1.15`. However, the issue https://github.com/prometheus-operator/kube-prometheus/issues/493 asked 1.14. Should I bump `1.15`?

Comment (kakkoyun) [reply .github/workflows/ci.yaml:6]: I think it's safe to bump directly to `1.15`. It's already a stable release. However, this is not a decision for me to make 😅 Let's give a chance to maintainers to review.

Comment (paulfantom) [reply .github/workflows/ci.yaml:6]: Let's bump to 1.15 ;)

**Comment by aweris** [comment]
Recreated PR in https://github.com/prometheus-operator/kube-prometheus/pull/787 because of the branch name change

**Pull request closed without merging**